### ISSUE TITLE
Roslyn 4.5 does not enable DiagnosticAnalyzers via the ruleset file anymore

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [dotnet/roslyn#67211]
**Version**: MSBuild version 17.5.0+6f08c67f3 - Roslyn 4.5

In [sonar-dotnet](https://github.com/SonarSource/sonar-dotnet/tree/master) we have a set of analyzers enabled by default and a set of analyzers that need to be explicitly enabled by a `ruleset` file. Enabling by `ruleset` had been working as expected until the latest release of MSBuild 17.5.

When our analyzers are used with MSBuild17.5 (Roslyn 4.5), all the rules not enabled by default are not executed, although they are enabled by a `.ruleset` file (which should have higher priority than the default values).

During debugging - assuming we have enabled a suppressed analyzer with `DiagnosticId`= "S126" through a ruleset file:
1. When inspecting the `Compilation.Options.SpecificDiagnosticOptions`, the analyzer seemed enabled with value `{[S126, Warn]}`; however, it was not executed.
2. When the value of `IsEnabledByDefault` is manually set to true [when creating the DiagnosticDescriptor object](https://github.com/SonarSource/sonar-dotnet/blob/dffe8ff1aaeffb92431d5fa5e811c07b00da4808/analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/DiagnosticDescriptorFactory.cs#L45), then S126 was normally executed.
3. When the analyzer S126 is enabled through `.editorconfig,` it works as expected.
4. When we downgrade Roslyn to 4.4 - the ruleset enabling works as expected.

The analyzers enabled by default are still properly disabled by the ruleset file.

In summary - Roslyn does not take into account analyzers enabled through a ruleset file.
I'll gladly provide any log or other info to help with the investigation.

**Comment by mary-georgiou-sonarsource** [comment]
Reproducer:
The project below is referencing the sonar-dotnet analyzers and I have enabled S126 via the `custom.ruleset` file.
```bash
git clone https://github.com/mary-georgiou-sonarsource/roslyn-bug-67211-repro
cd roslyn-bug-67211-repro
# Using MSBuild  17.5.0+6f08c67f3
MSBuild.exe /t:Clean,Build
```

Issues in the output:
```
warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to a
wait non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. 
warning S1118: Add a 'protected' constructor or the 'static' keyword to the class declaration.
warning S3168: Return 'Task' instead.
warning S1186: Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation. 
```

Then re-build with an older MSBuild Version
```bash
# Build Engine version 17.2.1+52cd2da31
MSBuild.exe /t:Clean,Build
```

And the issues are: 
```
warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to a
wait non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. 
warning S1118: Add a 'protected' constructor or the 'static' keyword to the class declaration..
warning S3168: Return 'Task' instead. 
warning S1186: Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation
warning S126: Add the missing 'else' clause with either the appropriate action or a suitable comment as to why no action is taken. 
```

S126 is not reported with the newest Roslyn although its enabled via the `custom.ruleset` file.

**Comment by davidwengier** [comment]
@mavasani

**Comment by mavasani** [comment]
Fix is in process of being ported to 17.5 servicing:  https://github.com/dotnet/roslyn/pull/67300

**Comment by 333fred** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/66085.

**Comment by mary-georgiou-sonarsource** [comment]
@mavasani when will this fix be shipped?
